### ISSUE TITLE
Issue with using HMS branch on a GMS + HMS mobile device

### ISSUE THREAD

**Issue by suporteproside** [transistorsoft/react-native-background-geolocation#1658]
## Your Environment
* Plugin version: https://github.com/transistorsoft/react-native-background-geolocation.git#hms
* Platform: Android
* OS version: 13 TKQ1.220829.002
* Device manufacturer / model: POCO F3 - M2012K11AG
* React Native version (`react-native -v`): 0.70.6
* Plugin config: 
```javascript <!-- Syntax-highlighting:  paste your code below -->

    const readyConfig: Config = {
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      stationaryRadius: 25,
      disableLocationAuthorizationAlert: true,
      showsBackgroundLocationIndicator: true,
      distanceFilter: 20,
      heartbeatInterval: 60,
      preventSuspend: true,
      stopTimeout: 5,
      debug: false, // debug sound
      foregroundService: true,
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      stopOnTerminate: false,
      startOnBoot: true,
      enableHeadless: true,
      batchSync: false,
      autoSync: false,
      locationAuthorizationRequest: 'WhenInUse',
      locationAuthorizationAlert: locationAuthorizationAlert,
      backgroundPermissionRationale: backgroundPermission,
      notification: foregroundNotification,
    };

```

## Expected Behavior
Background geolocation events and method should run without an issue.

## Actual Behavior
Bakcground geolocation location getCurrentPosition and (possibly) other methods do not run. See images below.

## Steps to Reproduce
1. Make sure you're using the HMS branch.
2. Install HMS on a GMS device.
3. Build and install the app on said device. 
4. Error will occur.

## Context
Was running the HMS branch on a GMS device that also happened to have HMS for other reasons. 

Although the "problem" got fixed **after** I uninstalled HMS, it seems like it might cause an issue on devices with HMS + GMS (?), it might need further testing.

## Debug logs
![image](https://user-images.githubusercontent.com/39157662/226597088-02251862-f6d7-4801-b2b6-3bdb1420dd2b.png)
![image](https://user-images.githubusercontent.com/39157662/226597565-486862a3-f1ae-4f7c-8a92-ea1f80e0dd84.png)

![image](https://user-images.githubusercontent.com/39157662/226597653-d4e00ab2-297e-4310-837c-89673ac7a269.png)

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
PASTE_YOUR_LOGS_HERE
```

</details>

TLDR: Issue was found while using the HMS branch on a device with both GMS and HMS. After uninstalling HMS it works fine. Branch works well on a HMS only device and GMS only device.

**Comment by christocracy** [comment]
Thanks for the feedback.  I pushed a new commit to `hms` branch to prefer `GMS` when both `HMS` and `GMS` are detected.

**Comment by suporteproside** [comment]
You welcome. Will keep testing it and report if anything comes up.
